### ISSUE TITLE
Issue with Multi-Configuration build

### ISSUE THREAD

**Issue by DerekDally** [elvanja/jenkins-gitlab-hook-plugin#50]
Hi Vanja - 

Is there a way to prevent the plugin from explicitly triggering both the upstream and downstream builds of a multi-configuration project?  For example, if I have a job named 'foo', which has 'debug' and 'release' configurations, 'foo' will be triggered, as will 'foo/BUILD=debug' and 'foo/BUILD=release'.  However, foo itself will also trigger these jobs, resulting in queued duplicate jobs.

For now, I changed the code to return false in matches? (project.rb) if self includes a slash, though I doubt this is the most robust way of handling this.  Any ideas?  Is this a configuration I'm missing?

Thanks,
Derek

**Comment by nowtv-web** [comment]
Hi, I know this has been close already but I'm facing similar problem however I don't use multi-configuration.

Let say there is a parent build which should be triggered by the hook named _foo_ and a few downstream jobs, let say _bar-1_, _bar-2_, and _bar-3_.

Now the _foo_ job can be a regular job or multiphase, it doesn't actually matter. It's duty is to get triggered by the hook and then run the downstream jobs in a sequence. Here the problem starts. The parent job _foo_ gets triggered by the hook as it should but the the _bar-*_ jobs are triggered by the hook as well (and by the parent too). Which results in a downstream jobs being triggered twice and not really in a sequence.

The question is, how to prevent 'trigger sensitiveness' of the downstream _bar-*_ jobs while keep them git repo aware so they could do the checkout?

Here is [related question on Stack](http://stackoverflow.com/questions/34953073/jenkins-and-gitlab-how-to-setup-scm-aware-job-which-is-not-triggered-by-the-hoo) I posted.

Perhaps the better idea is to create a new issue but this seems kind of related and could have been already solved (the GitLab plugin on the Jenkins instance I'm using is most likely outdated).

**Comment by topr** [comment]
Hi, I know this has been closed already but I'm facing similar problem, I don't use multi-configuration though.

Let say there is a parent build named _foo_ which should be triggered by the hook and a few downstream jobs, let say _bar-1_, _bar-2_, and _bar-3_.

Now the _foo_ job can be a regular job or multiphase, it doesn't actually matter. Its duty is to get triggered by the hook and then run the downstream jobs in a sequence. Here the problem starts. The parent job _foo_ gets triggered by the hook as it should but the the _bar-*_ jobs are triggered directly by the hook as well and by the parent as well. Which results in a downstream jobs being triggered twice and not really in a sequence.

The question is, how to prevent 'trigger sensitiveness' of the downstream _bar-*_ jobs while keep them git repo aware so they could do the checkout?

Here is [related question on Stack](http://stackoverflow.com/questions/34953073/jenkins-and-gitlab-how-to-setup-scm-aware-job-which-is-not-triggered-by-the-hoo) I posted.

Perhaps the better idea is to create a new issue but this seems kind of related and could have been already solved (the GitLab plugin on the Jenkins instance I'm using is most likely outdated).

EDIT: it looks I have found the answer here: https://github.com/elvanja/jenkins-gitlab-hook-plugin/issues/11#issuecomment-35385032 :+1:
